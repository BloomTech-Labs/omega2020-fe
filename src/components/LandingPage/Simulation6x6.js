import React from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Game6x6 from '../../images/Game6x6.png';
import PurpleButton from '../assets/PurpleButton';

const Simulation6x6 = (props) => {
  const classes = useStyles();

  let viewPort = false;
  const md = useMediaQuery('(min-width: 1050px)');
  if (viewPort === md) {
    viewPort = true;
  } else {
    viewPort = false;
  }

  return (
    <Box className={classes.root}>
      <Box className={classes.mainContainer}>
        {(() => {
          switch (viewPort) {
            case true:
              return (
                <Box style={{ textAlign: 'center' }}>
                  <Typography
                    className={classes.header}
                    style={{ fontSize: '1.3rem' }}
                    id='solve'
                  >
                    Try to solve this 6x6 sudoku:
                  </Typography>
                  <br />
                  <Typography className={classes.subheader} id='row'>
                    Each row, column and square needs to be filled
                    <br />
                    out with the numbers 1-6, without repeating any
                    <br />
                    numbers within the row, column or square.
                  </Typography>
                  <br />
                  <Box>
                    <img src={Game6x6} alt='Game6x6' width='300' />
                  </Box>
                </Box>
              );
            default:
              return (
                <Box className={classes.container}>
                  <Box className={classes.rightContainer}>
                    <Typography
                      className={classes.header}
                      style={{ fontSize: '1.3rem' }}
                    >
                      Try to solve this 6x6 sudoku:
                    </Typography>
                    <br />
                    <Typography className={classes.subheader}>
                      Each row, column and square needs to be filled out with
                      the numbers 1-6,
                      <br />
                      without repeating any numbers within the row, column or
                      square.
                    </Typography>
                    <br />
                    <Typography
                      className={classes.header}
                      style={{ fontSize: '1.3rem' }}
                    >
                      Instructions:
                    </Typography>
                    <br />
                    <Typography className={classes.subheader}>
                      In the light blue box enter the number 6.
                    </Typography>
                    <br />
                    <Typography
                      className={classes.header}
                      style={{ fontSize: '1.3rem' }}
                    >
                      Do you know what comes next?
                    </Typography>
                    <br />
                    <Typography className={classes.subheader}>
                      Try to complete the restâ€¦ you got this!
                    </Typography>
                    <br />
                    <Box className={classes.space}>
                      <Typography variant='caption'>Need more help?</Typography>
                      <PurpleButton title={'Tutorial Coming Soon'} />
                    </Box>
                  </Box>
                  <Box className={classes.leftContainer}>
                    {/* <img src={Game6x6} alt='Game6x6'/> */}
                  </Box>
                </Box>
              );
          }
        })()}
      </Box>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 100,
    marginBottom: 100,
  },
  mainContainer: {
    width: '100%',
    display: 'flex',
    flexFlow: 'row wrap',
    justifyContent: 'space-evenly',
    alignItems: 'center',
  },
  subheader: {
    // opacity: 0.3,
    color: '#A2A2A2',
    display: 'flex',
    flexFlow: 'row wrap',
  },
  container: {
    width: '100%',
    display: 'flex',
    flexFlow: 'row wrap',
    justifyContent: 'space-evenly',
    alignItems: 'center',
  },
  leftContainer: {
    width: '500px',
    height: '400px',
    backgroundImage: `url(${Game6x6})`,
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'contain, cover',
    backgroundPosition: 'center',
  },
  rightContainer: {
    width: '40%',
  },
  space: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
}));

export default Simulation6x6;
