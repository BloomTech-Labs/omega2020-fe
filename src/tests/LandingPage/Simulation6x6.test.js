import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Typography } from '@material-ui/core';
import Game6x6 from '../../images/Game6x6.png';
import Simulation6x6 from '../../components/LandingPage/Simulation6x6';

configure({ adapter: new Adapter() });

describe('the information on the simulation', () => {
  const simulation6x6Render = renderer.create(<Simulation6x6 />); 
  const simulation6x6Root = simulation6x6Render.root;
  
  const wrapper = mount(<Simulation6x6 />);
  const iWrapper = wrapper.instance();
  
  it('snapshot the info on the Game Simulation', () => {
    const simulation = renderer.create(<Simulation6x6 />).toJSON();
    expect(simulation).toMatchSnapshot();
  });

  it('checks the image', () => {
      expect(wrapper.find('img').prop('src')).toEqual(Game6x6)
  })

  it('checks the root for info (props and type)', () => {
    expect(wrapper.containsMatchingElement(
    <Typography>
        Try to solve this 6x6 sudoku:
    </Typography>
    )).toBeTruthy();

    expect(wrapper.containsMatchingElement(
    <Typography>
        Each row, column and square needs to be filled
        <br />
        out with the numbers 1-6, without repeating any
        <br />
        numbers within the row, column or square.
    </Typography>
    )).toBeTruthy();
    


    expect(iWrapper).toEqual(null)
    expect(simulation6x6Root.findByProps({id: 'solve'}).props.children).toBeTruthy();
    expect(simulation6x6Root.findByProps({id: 'row'}).props.children).toBeTruthy();    

  });

});
