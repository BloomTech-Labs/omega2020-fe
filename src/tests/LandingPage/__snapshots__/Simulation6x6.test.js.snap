// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the information on the simulation snapshot the info on the Game Simulation 1`] = `
<div
  className="MuiBox-root MuiBox-root-16 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-17 makeStyles-mainContainer-2"
  >
    <div
      className="MuiBox-root MuiBox-root-18"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        id="solve"
        style={
          Object {
            "fontSize": "1.3rem",
          }
        }
      >
        Try to solve this 6x6 sudoku:
      </p>
      <br />
      <p
        className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
        id="row"
      >
        Each row, column and square needs to be filled
        <br />
        out with the numbers 1-6, without repeating any
        <br />
        numbers within the row, column or square.
      </p>
      <br />
      <div
        className="MuiBox-root MuiBox-root-19"
      >
        <img
          alt="Game6x6"
          src="Game6x6.png"
          width="300"
        />
      </div>
    </div>
  </div>
</div>
`;
